src/zod.ts

```diff
@@ -1,231 +1,211 @@
+			invalid_type_error:
+				"skipRecheck must be a defined boolean (true or false)",
+		}),
+		maxDataDepth: z.number().gte(1, {
+			message: "maxDataDepth must be a number greater than 0",
+		}),
+		port: z.number().gte(1).lte(65535, {
+			message: "port must be a number between 1 and 65535",
+		}),
+		searchLimit: z.union([z.number(), z.undefined()], {
+			invalid_type_error: "searchLimit must be a number or undefined",
+		}),
+		delay: z.number().gt(1).lt(1000, {
+			message:
+				"delay is in seconds, >1000 implies you think it's milliseconds",
+		}),
+		action: z.nativeEnum(Action, {
+			invalid_type_error: `action must be either '${Action.SAVE}' or '${Action.INJECT}'`,
+		}),
+		qbittorrentUrl: z.string().url().optional(),
+		rtorrentRpcUrl: z.string().url().optional(),
+		transmissionRpcUrl: z.string().url().optional(),
 
-			if (Array.isArray(values)) {
-				return values.every(validatePath);
-			} else {
-				return validatePath(values);
-			}
-		},
-		{
-			message: "one or more paths do not exist on the filesystem",
-		}
-	),
-	BOOLEAN: z
-		.boolean()
-		.refine((value) => value !== undefined && value !== null, {
-			message: "value must be a defined boolean (true or false)",
+		matchMode: z.nativeEnum(MatchMode, {
+			invalid_type_error: `matchMode must be either '${MatchMode.RISKY}' or '${MatchMode.SAFE}'`,
 		}),
-	PORT: z.number().refine((value) => value >= 1 && value <= 65535, {
-		message: "port must be a number between 1 and 65535",
-	}),
-	NUMBER: z.union([z.number(), z.undefined()]).refine(
-		(value) => {
-			return typeof value === "number" || value === undefined;
-		},
-		{
-			message: "value must be a number or undefined",
-		}
-	),
-	DELAY: z.number().refine((value) => value >= 1 && value <= 1000, {
-		message:
-			"delay is in seconds, >1000 implies you think it's milliseconds",
-	}),
-	ACTION: z
-		.string()
-		.refine((value) => value === Action.SAVE || value === Action.INJECT, {
-			message: `value must be either '${Action.SAVE}' or '${Action.INJECT}'`,
+		linkType: z.nativeEnum(LinkType, {
+			invalid_type_error: `linkType must have a value of "${LinkType.HARDLINK}" or "${LinkType.SYMLINK}"`,
+		}),
+		dataCategory: z.union([
+			z.undefined(),
+			z.string().min(1).max(24, {
+				message: "dataCategory must have a length of 1-24 characters",
+			}),
+		]),
+		fuzzySizeThreshold: z.number().gt(0).lt(1, {
+			message:
+				"fuzzySizeThreshold must be a decimal percentage greater than 0 and less than 1",
 		}),
-	MATCHMODE: z
-		.enum([MatchMode.SAFE, MatchMode.RISKY])
-		.refine(
-			(value) => value === MatchMode.RISKY || value === MatchMode.SAFE,
-			{
-				message: `value must be either '${MatchMode.RISKY}' or '${MatchMode.SAFE}'`,
+		searchCadence: z.union([
+			z.number(),
+			z.undefined(),
+			z.nan().refine(
+				() => {
+					return typeof fileConfig.searchCadence === "undefined";
+				},
+				{
+					message:
+						"your searchCadence does not follow vercel's `ms` style",
+				}
+			),
+		]),
```

Comment:
Doesn't it just work as:
```
		searchCadence: z.number().optional(),
```

Reply:
We need to allow for the use of `null` as well, so the solution according to the docs is `.nullish()`, actually.